### PR TITLE
SNOW-1843504 - Testing resubmitting requests using request id

### PR DESCRIPTION
**Pull request by sfc-gh-pmotacki**
### Description
Node.js - Testing resubmitting requests using request id

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [ ] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: test/integration/testUtil.js [hunk 1]
```diff
@@ -96,6 +96,18 @@ const executeCmdAsync = function (connection, sqlText, binds = undefined) {
 
 module.exports.executeCmdAsync = executeCmdAsync;
 
+const executeCmdAsyncWithMoreParameters = function (connection, sqlText, moreParameters) {
```
Comment (sfc-gh-pfus) [inline test/integration/testUtil.js:99]: Maybe `additionalParameters`? WDYT?

**Review by sfc-gh-pfus — APPROVED** [review]

File: test/integration/testExecute.js [hunk 2]
```diff
@@ -431,5 +431,88 @@ describe('Execute test - variant', function () {
 
     it(testCase.name, createItCallback(testCase, rowAsserts));
   });
-});
 
+  describe.only( 'connection.execute() Resubmitting requests using requestId and different connections', function () {
```
Comment (sfc-gh-dprzybysz) [inline test/integration/testExecute.js:435]: we should not keep only here?

File: test/integration/testExecute.js [hunk 3]
```diff
@@ -431,5 +431,88 @@ describe('Execute test - variant', function () {
...
+    const createTable = 'create or replace table test_request_id(colA string)';
+    let  firstConnection;
+    let  secondConnection;
+    before(async () => {
+      firstConnection = testUtil.createConnection();
+      secondConnection = testUtil.createConnection();
+      await testUtil.connectAsync(firstConnection);
+      await testUtil.connectAsync(secondConnection);
+      await testUtil.executeCmdAsync(firstConnection, 'truncate table if exists test_request_id');
```
Comment (sfc-gh-dprzybysz) [inline test/integration/testExecute.js:444]: do we need truncate just before create?

File: test/integration/testExecute.js [hunk 4]
```diff
@@ -431,5 +431,88 @@ describe('Execute test - variant', function () {
...
+      await testUtil.executeCmdAsync(firstConnection, createTable);
+    });
+
+    beforeEach(async () => {
+      await testUtil.executeCmdAsync(firstConnection, 'truncate table if exists test_request_id');
+    });
+
+    after(async () => {
+      await testUtil.executeCmdAsync(firstConnection, 'drop table if exists test_request_id');
```
Comment (sfc-gh-dprzybysz) [inline test/integration/testExecute.js:453]: should we also destroy connections?

File: test/integration/testExecute.js [hunk 5]
```diff
@@ -431,5 +431,88 @@ describe('Execute test - variant', function () {
...
+    });
+
+    it('Do not INSERT twice when the same request id and connection', async () => {
+      let requestId;
+      await testUtil.executeCmdAsyncWithAdditionalParameters(firstConnection, 'INSERT INTO test_request_id VALUES (\'testValue\');')
+        .then((result) => {
```
Comment (sfc-gh-dprzybysz) [inline test/integration/testExecute.js:459]: why do we mix await with promises? can we have these tests 100% async-await?

File: test/integration/testExecute.js (hunk 2 above)
Comment (sfc-gh-pmotacki) [reply test/integration/testExecute.js:435]: Of course. Removed

File: test/integration/testExecute.js (hunk 3 above)
Comment (sfc-gh-pmotacki) [reply test/integration/testExecute.js:444]: It isn't obligatory when we execute "create or replace" in the next step. Thank you

File: test/integration/testExecute.js (hunk 5 above)
Comment (sfc-gh-pmotacki) [reply test/integration/testExecute.js]: Sure. Fixed

File: test/integration/testExecute.js (hunk 4 above)
Comment (sfc-gh-pmotacki) [reply test/integration/testExecute.js:452]: Fixed

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Pull request merged**